firebase-web/src/main/java/io/spine/web/firebase/subscription/TimedSubscriptionMixin.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.web.firebase.subscription;
+
+import com.google.protobuf.Timestamp;
+import io.spine.annotation.GeneratedMixin;
+import io.spine.base.Time;
+import io.spine.client.SubscriptionId;
+import io.spine.client.Topic;
+import io.spine.client.TopicId;
+
+import static com.google.protobuf.util.Timestamps.compare;
+
+/**
+ * A generated mixin for the {@link TimedSubscription} type.
+ */
+@GeneratedMixin
+interface TimedSubscriptionMixin extends TimedSubscriptionOrBuilder {
+
+    /**
+     * Obtains the subscription ID.
+     */
+    default SubscriptionId id() {
+        return getSubscription().getId();
+    }
+
+    /**
+     * Obtains the topic ID.
+     */
+    default TopicId topicId() {
+        return topic().getId();
+    }
+
+    /**
+     * Obtains the subscription topic.
+     */
+    default Topic topic() {
+        return getSubscription().getTopic();
+    }
+
+    /**
+     * Checks if this subscription is expired.
+     *
+     * @return {@code true} if the subscription is expired and should be inactivated,
+     *         {@code false} if the subscription is still relevant
+     */
+    default boolean isExpired() {
+        Timestamp validThru = getValidThru();
+        Timestamp now = Time.currentTime();
+        return compare(now, validThru) > 0;
```

Comment:
Someone should explain this `>` (and not `>=`).

Ideally, that could be somewhere in the Protobuf definitions. But I cannot see any changes on that.

Reply:
I've added notes about when does the time elapse in Java and in Protobuf.
It is also hinted in the name "valid thru", (not "valid until").

It might be important for the users to understand this detail for the full picture, but, in practice, I don't think it matters much if the subscriptions become stale at the moment or after the moment. Unlike, say, delivery, this kind of granularity should not be expected, as the staleness is validated at the server but the prolongation requests are created at the client. I doubt there will be a case when this _really_ matters to the API user.